src/fal/cli/selectors.py

```diff
@@ -58,26 +59,57 @@ def create_plan_from_graph(cls, parsed, nodeGraph: NodeGraph, fal_dbt: FalDbt):
         return cls(list(set(ids_to_execute)), fal_dbt.project_name)
 
 
+@dataclass
+class SelectionUnion:
+    components: List[str]
+
+
+@dataclass
+class SelectionIntersection:
+    components: List[str]
+
+
+def parse_union(
+    components: List[str],
+) -> SelectionUnion:
+    # Based on the original implemention at dbt-core.
+
+    # turn ['a b', 'c'] -> ['a', 'b', 'c']
+    raw_specs = itertools.chain.from_iterable(r.split(OP_SET_UNION) for r in components)
+    union_components = []
+
+    # ['a', 'b', 'c,d'] -> union('a', 'b', intersection('c', 'd'))
+    for raw_spec in raw_specs:
+        union_components.append(
+            SelectionIntersection(
+                raw_spec.split(OP_SET_INTERSECTION),
+            )
+        )
+    return SelectionUnion(
+        union_components,
+    )
+
+
 def _filter_node_ids(
     unique_ids: List[str],
     fal_dbt: FalDbt,
-    selected_nodes: List[str],
+    selectors: List[str],
     nodeGraph: NodeGraph,
 ) -> List[str]:
     """Filter list of unique_ids according to a selector."""
-    output = []
+    output = set()
 
-    for selector in selected_nodes:
-        selector_plans = map(
-            lambda sub_selector: SelectorPlan(sub_selector, unique_ids, fal_dbt),
-            SET_INTERSECTION.separate(selector),
-        )
-        selector_outputs = map(
-            lambda plan: set(plan.execute(nodeGraph)), selector_plans
-        )
-        output.extend(set.intersection(*selector_outputs))
+    union = parse_union(selectors)
+    for intersection in union.components:
+        plan_outputs = [
+            set(SelectorPlan(selector, unique_ids, fal_dbt).execute(nodeGraph))
+            for selector in intersection.components
+            if selector
+        ]
+        if plan_outputs:
+            output |= set.intersection(*plan_outputs)
```

Comment:
What's happening here?

Reply:
We are taking the intersection of outputs, and then union'ing it into the existing result set. For example in the case of `a,b c,d`, we'll first take the intersection of `a` and `b` and then union this with the existing result set (empty set). And then in the second iteration we'll take the intersection of `c` and `d` and union this with the existing result set (intersection of `a` and `b`) which would give us `a,b` + `c,d`.

Reply:
Thanks for clarification.